browser/extensions/api/brave_rewards_api.cc

```diff
@@ -282,45 +282,126 @@ ExtensionFunction::ResponseAction BraveRewardsTipSiteFunction::Run() {
 
   // Sanity check: don't allow tips in private / tor contexts,
   // although the command should not have been enabled in the first place.
   if (!brave::IsRegularProfile(browser_context())) {
-    return RespondNow(
-        Error("Cannot tip Twitter user in a private context"));
+    return RespondNow(Error("Cannot tip user in a private context"));
   }
 
   Profile* profile = Profile::FromBrowserContext(browser_context());
   auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
-  if (rewards_service) {
-    AddRef();
-    std::map<std::string, std::string> args;
-    args["user_id"] = params->media_meta_data.user_id;
-    args["twitter_name"] = params->media_meta_data.twitter_name;
-    args["screen_name"] = params->media_meta_data.screen_name;
-    rewards_service->SaveInlineMediaInfo(
-        params->media_meta_data.media_type,
-        args,
-        base::Bind(&BraveRewardsTipTwitterUserFunction::
-                   OnTwitterPublisherInfoSaved,
-                   weak_factory_.GetWeakPtr()));
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
   }
 
+  AddRef();
+
+  rewards_service->GetPublisherInfo(
+      params->publisher_key,
+      base::Bind(
+          &BraveRewardsTipUserFunction::OnTipUserGetPublisherInfo,
+          this));
+
   return RespondNow(NoArguments());
 }
 
+void BraveRewardsTipUserFunction::OnTipUserGetPublisherInfo(
+    const ledger::type::Result result,
+    ledger::type::PublisherInfoPtr info) {
+  if (result != ledger::type::Result::LEDGER_OK &&
+      result != ledger::type::Result::NOT_FOUND) {
+    Release();
+    return;
+  }
+
+  if (result == ledger::type::Result::LEDGER_OK) {
+    ShowTipDialog();
+    Release();
+    return;
+  }
+
+  std::unique_ptr<brave_rewards::TipUser::Params> params(
+      brave_rewards::TipUser::Params::Create(*args_));
+
+  auto publisher_info = ledger::type::PublisherInfo::New();
+  publisher_info->id = params->publisher_key;
+  publisher_info->name = params->publisher_name;
+  publisher_info->url = params->url;
+  publisher_info->provider = params->media_type;
+  publisher_info->favicon_url = "";
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  auto* rewards_service = RewardsServiceFactory::GetForProfile(profile);
+  if (!rewards_service) {
+    Release();
+    return;
+  }
+
+  rewards_service->SavePublisherInfo(
+      0,
+      std::move(publisher_info),
+      base::Bind(&BraveRewardsTipUserFunction::
+                 OnTipUserSavePublisherInfo,
+                 weak_factory_.GetWeakPtr()));
+}
+
+void BraveRewardsTipUserFunction::OnTipUserSavePublisherInfo(
+    const ledger::type::Result result) {
+  if (result != ledger::type::Result::LEDGER_OK) {
+    Release();
+    return;
+  }
+
+  ShowTipDialog();
+  Release();
+}
+
+void BraveRewardsTipUserFunction::ShowTipDialog() {
+  std::unique_ptr<brave_rewards::TipUser::Params> params(
```

Comment:
is it possible that this is null? if so please add null check

Reply:
Fixed.